tests/lib/compose-features.py

```diff
@@ -0,0 +1,235 @@
+    replaced by the rerun data from the new_json_file
+    '''
+    with open(old_json_file, 'r') as f:
+        old_json = json.load(f)
+    with open(new_json_file, 'r') as f:
+        new_json = json.load(f)
+    for test in new_json['tests']:
+        for old_test in old_json['tests']:
+            if old_test['task-name'] == test['task-name'] and old_test['suite'] == test['suite'] and old_test['variant'] == test['variant']:
+                old_test.clear()
+                for key, value in test.items():
+                    old_test[key] = value
+                break
+    return old_json
+
+
+def replace_old_runs(dir, output_dir):
+    '''
+    Given the directory in input (dir) that contains a set of files of original
+    run data together with rerun data, this populates the specified output_dir
+    with a consolidated set of composed features, one per system. An original
+    composed features file is a file that ends in _1.json. A rerun composed
+    features file is a file that ends in _<num>.json where <num> is greater 
+    than 1. The numbering is automatically generated when the compose features
+    script was called with the --run-attempt
+
+
+    :param dir: directory containing composed feature files with varying run 
+    attempt numbers
+    :param output_dir: directory where to write the consolidated composed features
+    '''
+    os.makedirs(output_dir)
+    filenames_no_ext = [os.path.splitext(f)[0] for f in os.listdir(
+        dir) if os.path.isfile(os.path.join(dir, f))]
+    reruns_no_ext = [
+        file for file in filenames_no_ext if not file.endswith('_1')]
+    originals_no_ext = [file for file in filenames_no_ext if file.endswith(
+        '_1') and any(rerun for rerun in reruns_no_ext if rerun.startswith(file[:-2]))]
+    reruns_no_ext.sort(key=lambda x: int(x.split('_')[-1]))
+    for rerun in reruns_no_ext:
+        beginning = '_'.join(rerun.split('_')[:-1])
+        original = list(
+            filter(lambda x: x.startswith(beginning), originals_no_ext))
+        if len(original) != 1:
+            raise RuntimeError(
+                "The rerun %s does not have a corresponding original run" % rerun)
+        tests = _replace_tests(os.path.join(
+            dir, original[0] + ".json"), os.path.join(dir, rerun + ".json"))
+        with open(os.path.join(output_dir, beginning + ".json"), 'w') as f:
+            f.write(json.dumps(tests))
+    for file in filenames_no_ext:
+        if file not in originals_no_ext:
+            shutil.copyfile(os.path.join(dir, file + ".json"),
+                            os.path.join(output_dir, '_'.join(file.split('_')[:-1]) + '.json'))
+
+
+def run_attempt_type(value):
+    if value is not int or int(value) <= 0:
+        raise argparse.ArgumentTypeError(
+            "%s is invalid. Run attempts are integers and start at 1" % value)
+    return value
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="""
```

Comment:
Maybe move the description variable above main to make this format with less indent?